evalml/pipelines/components/transformers/preprocessing/featuretools.py

```diff
@@ -42,6 +44,13 @@ def _make_entity_set(self, X):
             es = ft_es.add_dataframe(dataframe=X, dataframe_name="X", index=self.index)
         return es
 
+    def _should_skip_transform(self, X):
+        for feature in self.features:
+            feature_names = feature.get_feature_names()
+            if not set(feature_names).issubset(set(X.columns)):
```

Comment:
Should we be giving the user a warning with this? What are scenarios where this would be false?

Reply:
Should this be `set(feature_names).intersection(set(X.columns))` ? Is there a case where users dropped some  features in between running dfs and modeling?

Reply:
Cause right now we skip only if all the `feature_names` are in `X`. But if 3 out of the 4 features are present, we'll get a duplicate column name error right?

Reply:
@freddyaboulton thanks for the example, I'll rework the logic here to take that case into account.